clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -689,6 +689,9 @@ private void handleListOffsetResponse(Map<TopicPartition, Long> timestampsToSear
                     parsedRecords = new PartitionRecords<>(fetchOffset, tp, parsed);
                     ConsumerRecord<K, V> record = parsed.get(parsed.size() - 1);
                     this.sensors.recordsFetchLag.record(partition.highWatermark - record.offset());
+                } else if (partition.records.iterator().hasNext()) {
```

Comment:
@hachikuji Sorry for late reply. Sure, I will replace `position` with `fetchOffset` in the patch. I still keep the existing measurement of fetchLag in the case where the response is not empty because I don't want to cause any unexpected problem by making this subtle but backward incompatible change. On the other hand I don't mind if you or someone else want to change it.